k210.svd

```diff
@@ -2728,12 +2728,74 @@
             <groupName>SHA256</groupName>
             <baseAddress>0x502C0000</baseAddress>
             <registers>
-                <!-- TODO -->
                 <register>
-                    <name>dummy</name>
-                    <description>Dummy register: this peripheral is not implemented yet</description>
+                    <dim>8</dim>
+                    <dimIncrement>0x4</dimIncrement>
+                    <name>sha_result[%s]</name>
+                    <description>Calculated SHA256 return value</description>
                     <addressOffset>0x00</addressOffset>
                 </register>
+                <register>
+                    <name>sha_data_in1</name>
+                    <description>SHA256 input data from this register</description>
+                    <addressOffset>0x20</addressOffset>
+                </register>
+                <register>
+                    <name>sha_num_reg</name>
+                    <description>Counters register</description>
+                    <addressOffset>0x28</addressOffset>
+                    <fields>
+                        <field>
+                            <name>sha_data_cnt</name>
+                            <description>The total amount of data calculated by SHA256 is set by this register, and the smallest unit is 512bit</description>
+                            <bitRange>[15:0]</bitRange>
+                        </field>
+                        <field>
+                            <name>sha_data_num</name>
+                            <description>Currently calculated block number. 512bit=1block</description>
+                            <bitRange>[31:16]</bitRange>
+                        </field>
+                    </fields>
+                </register>
+                <register>
+                    <name>sha_function_reg_0</name>
+                    <description>Function configuration register 0</description>
+                    <addressOffset>0x2C</addressOffset>
+                    <fields>
+                        <field>
+                            <name>sha_en</name>
+                            <description>write:SHA256 enable register. read:Calculation completed flag</description>
+                            <bitRange>[0:0]</bitRange>
+                        </field>
+                        <field>
+                            <name>sha_overflow</name>
+                            <description>SHA256 calculation overflow flag</description>
+                            <bitRange>[8:8]</bitRange>
+                        </field>
+                        <field>
+                            <name>sha_endian</name>
+                            <description>Endian setting; 0:little endian 1:big endian</description>
```

Comment:
does it make sense to use an enumeration here?

Reply:
Yes.